thirdparty/runtime_environments.go

```diff
@@ -347,6 +367,88 @@ func stringToTime(timeInitial string) (time.Time, error) {
 	return time.Unix(timestamp, 0), nil
 }
 
+// Image stores information about an image including its name, version_id, kernel, ami, and its last deployed time.
+type Image struct {
+	Name         string
+	VersionID    string
+	Kernel       string
+	LastDeployed time.Time
+	AMI          string
+}
+
+// getNameFromOSInfo uses the provided AMI and name (exact match) arguments to filter the image information.
+func (c *RuntimeEnvironmentsClient) getNameFromOSInfo(ctx context.Context, ami string, name string) (string, error) {
+	optsOS := OSInfoFilterOptions{
+		AMI:  ami,
+		Name: fmt.Sprintf("^%s$", name),
+	}
+	resultOS, err := c.GetOSInfo(ctx, optsOS)
+	if err != nil {
+		return "", errors.Wrap(err, "getting OS info")
+	}
+	if len(resultOS) == 0 {
+		return "", errors.Errorf("OS information name '%s' not found for distro", optsOS.Name)
+	}
+	if len(resultOS) > 1 {
+		return "", errors.Errorf("multiple options found for OS information name '%s'", optsOS.Name)
+	}
+	return resultOS[0].Version, nil
+}
+
+// GetImageInfo returns information about a image.
+func (c *RuntimeEnvironmentsClient) GetImageInfo(ctx context.Context, imageID string) (*Image, error) {
+	optsHistory := DistroHistoryFilterOptions{
+		Distro: imageID,
+		Limit:  1,
+	}
+	resultHistory, err := c.getHistory(ctx, optsHistory)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting history for image '%s': '%s'", imageID, err.Error())
+	}
+	if len(resultHistory) == 0 {
+		return nil, errors.Errorf("history for image '%s' not found", imageID)
+	}
+	if resultHistory[0].AMI == "" {
+		return nil, errors.Errorf("latest ami for image '%s' not found", imageID)
+	}
+	ami := resultHistory[0].AMI
+
+	// Determine timestamp field.
+	if resultHistory[0].CreationDate == "" {
+		return nil, errors.Errorf("creation time for image '%s' not found", imageID)
+	}
+	timestamp, err := stringToTime(resultHistory[0].CreationDate)
+	if err != nil {
+		return nil, errors.Wrap(err, "converting creation time: '%s'")
+	}
+
+	// Determine name field.
+	name, err := c.getNameFromOSInfo(ctx, ami, OSNameField)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting OSInfo for distro: '%s'", imageID)
```

Comment:
nit. wording and formatting
```suggestion
		return nil, errors.Wrapf(err, "getting OSInfo for image '%s'", imageID)
```